exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -47,12 +52,44 @@ class AwsDeployer
 
 
   nuke: ->
-    @terraform
-      ..get (err) ->
+    @terraform-file-builder.generate-provider-credentials!
+    process.stdout.write "terraform starting nuke from AWS"
+    @terraform.destroy (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting nuke from AWS"
-        ..destroy (err) ~>
-          | err => return process.stdout.write err.message
+        @_remove-hosted-zone @domain-name
+
+
+  _get-hosted-zone-id: (done) ->
+    @_hosted-zone-exists @domain-name, (id) ~>
+      if id then done id
+      else @_create-hosted-zone @domain-name, (id) -> done id
+
+
+  _hosted-zone-exists: (domain-name, done) ->
+    @route53 = new Aws.Route53 {api-version: '2013-04-01'}
+      ..list-hosted-zones null, (err, data) ~>
+        | err => return process.stdout.write err.message
```

Comment:
Let's follow the standard Node API for async APIs, instead of inventing our own convention. The callback is always called. Its first parameter is an error, the second optional return data.